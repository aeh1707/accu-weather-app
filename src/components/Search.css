.search {
  background-image: url('./kanchanara-fsSGgTBoX9Y-unsplash.jpg');
  background-size: cover;
  height: 300px;
  display: flex;
  justify-content: center;
  align-items: center;
}

input {
  min-width: 300px;
  font-size: 1.5rem;
  padding: 10px;
  border-radius: 50px;
}