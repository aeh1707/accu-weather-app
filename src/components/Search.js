import './Search.css';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { setSearchTerm, selectSearchTerm } from '../redux/searchReducer';

export default function Search() {
  const dispatch = useDispatch();
  const searchTerm = useSelector(selectSearchTerm);

  const onSearch = ({ target }) => {
    const { value } = target;
    dispatch(setSearchTerm(value));
  };

  return (
    <div className="search">
      <div className="search-container">
        <input
          id="search"
          type="text"
          value={searchTerm}
          onChange={onSearch}
          placeholder="ex: bitcoin"
        />
      </div>
    </div>
  );
}
