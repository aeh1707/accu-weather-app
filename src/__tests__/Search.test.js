/*eslint-disable*/
import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import store from '../redux/configureStore';
import Search from '../components/Search';

test('TopCoins component testing', () => {
  const TREE = TestRenderer.create(
    <Provider store={store}>
      <Search />
    </Provider>,
  );
  expect(TREE).toMatchSnapshot();
});
