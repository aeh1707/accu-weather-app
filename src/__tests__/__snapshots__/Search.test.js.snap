// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopCoins component testing 1`] = `
<div
  className="search"
>
  <div
    className="search-container"
  >
    <input
      id="search"
      onChange={[Function]}
      placeholder="ex: bitcoin"
      type="text"
      value=""
    />
  </div>
</div>
`;
